core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -257,10 +262,16 @@ public void startUp(Session session, String keyspace, boolean resetDb, boolean c
                 .put(STRING, Functions::getStringDataPoint)
                 .build();
 
+        metricsIdxCache = CacheBuilder.newBuilder().expireAfterAccess(1, TimeUnit.HOURS).maximumSize(40000).build();
```

Comment:
Distributed does not mean partitioned, it could be asynchronously replicated cache just as well. In that case the performance should be the same when it comes to local cache requests. The point is to reduce unnecessary writes to allow Cassandra to process more inserts when it comes to the datapoints.

Also, the cache should be populated in the start by requesting the metricIds from the Cassandra.